src/index.js

```diff
@@ -0,0 +1,41 @@
+import core, { getInput } from '@actions/core';
+import { getOctokit } from '@actions/github';
+
+(async () => {
+  const token = getInput('github-token', { required: true })
+  const checks = getInput('check', { required: true }).split("\n")
+  const repo = getInput('repo', { required: true })
+  const repoOwner = repo.split("/")[0]
+  const repoName = repo.split("/")[1]
+  const status = getInput('status', { required: true })
+  const branch = getInput('branch', {required: true })
+  const octokit = getOctokit(token)
+
+  const commitStatuses = await octokit.repos.listCommitStatusesForRef({
+    owner: repoOwner,
+    repo: repoName,
+    ref: branch,
```

Comment:
If I understand correctly, per https://octokit.github.io/rest.js/v20, `octokit.rest.repos.listCommitStatusesForRef` accepts either branch name or SHA for ref. So why not send the commit SHA instead, either this could be susceptible to catching the checks from the wrong commit? e.g. if commit A has the checks passing and then the checks for the more recent commit B are not started yet, this action would catch the stale checks for commit A.

Reply:
Solid point, we should be using param here.